### ISSUE TITLE
False postive on spaces around operators when functions are involved

### ISSUE THREAD

**Issue by pwildenhain** [alanmcruickshank/sqlfluff#207]
The linter complains about operators must be preceeeded by a space, but it seems like this criteria is being met?

Input:
```sql
select
    field,
    date(field_1) - date(field_2) as diff
from table
```
Output:
```
$ sqlfluff lint test.sql
== [test2.sql] FAIL
L:   3 | P:  21 | L006 | Operators should be preceded by a space.
```

This works just fine, if I take the functions away
```sql
select
    field,
    field_1 - field_2 as diff
from table
```

**Comment by pwildenhain** [comment]
I also tried adding spaces around the parentheses (even though looks a little weird to me) but that produced the same lint message:
`test.sql`
```sql
select
    field,
    date( field_1 ) - date( field_2 ) as diff
from table
```

```
$ sqlfluff lint test.sql
== [test.sql] FAIL
L:   3 | P:  21 | L006 | Operators should be preceded by a space.
```

**Comment by jonklippenstein-io** [comment]
I've run into the same thing with functions in a where clause when linting some existing models, e.g.: 

`SELECT * FROM table WHERE (a = func() OR a = func1())`

From the parser output it looks like FunctionSegment is picking up the whitespace as part of the function itself, rather than leaving it in front of the operator:

```
 FileSegment:                                                [0](1, 1, 1)
    StatementSegment:                                       [0](1, 1, 1)
        SelectStatementSegment:                             [0](1, 1, 1)
            SelectClauseSegment:                            [0](1, 1, 1)
                SELECT_KeywordSegment:                      [0](1, 1, 1)        'SELECT'
                whitespace_RawSegment:                      [6](1, 1, 7)        ' '
                [META] Indent:                              [7](1, 1, 8)
                SelectTargetElementSegment:                 [7](1, 1, 8)
                    star_KeywordSegment:                    [7](1, 1, 8)        '*'
                [META] Dedent:                              [8](1, 1, 9)
            whitespace_RawSegment:                          [8](1, 1, 9)        ' '
            FromClauseSegment:                              [9](1, 1, 10)
                FROM_KeywordSegment:                        [9](1, 1, 10)       'FROM'
                [META] Indent:                              [13](1, 1, 14)
                whitespace_RawSegment:                      [13](1, 1, 14)      ' '
                TableExpressionSegment:                     [14](1, 1, 15)
                    ObjectReferenceSegment:                 [14](1, 1, 15)
                        identifier_ReSegment:               [14](1, 1, 15)      'table'
                [META] Dedent:                              [19](1, 1, 20)
            whitespace_RawSegment:                          [19](1, 1, 20)      ' '
            WhereClauseSegment:                             [20](1, 1, 21)
                WHERE_KeywordSegment:                       [20](1, 1, 21)      'WHERE'
                [META] Indent:                              [25](1, 1, 26)
                whitespace_RawSegment:                      [25](1, 1, 26)      ' '
                ExpressionSegment:                          [26](1, 1, 27)
                    start_bracket_KeywordSegment:           [26](1, 1, 27)      '('
                    [META] Indent:                          [27](1, 1, 28)
                    ObjectReferenceSegment:                 [27](1, 1, 28)
                        identifier_ReSegment:               [27](1, 1, 28)      'a'
                    whitespace_RawSegment:                  [28](1, 1, 29)      ' '
                    equals_KeywordSegment:                  [29](1, 1, 30)      '='
                    whitespace_RawSegment:                  [30](1, 1, 31)      ' '
                    FunctionSegment:                        [31](1, 1, 32)
                        function_name_ReSegment:            [31](1, 1, 32)      'func'
                        start_bracket_KeywordSegment:       [35](1, 1, 36)      '('
                        end_bracket_KeywordSegment:         [36](1, 1, 37)      ')'
                        whitespace_RawSegment:              [37](1, 1, 38)      ' '
                    OR_KeywordSegment:                      [38](1, 1, 39)      'OR'
                    whitespace_RawSegment:                  [40](1, 1, 41)      ' '
                    ObjectReferenceSegment:                 [41](1, 1, 42)
                        identifier_ReSegment:               [41](1, 1, 42)      'a'
                    whitespace_RawSegment:                  [42](1, 1, 43)      ' '
                    equals_KeywordSegment:                  [43](1, 1, 44)      '='
                    whitespace_RawSegment:                  [44](1, 1, 45)      ' '
                    FunctionSegment:                        [45](1, 1, 46)
                        function_name_ReSegment:            [45](1, 1, 46)      'func1'
                        start_bracket_KeywordSegment:       [50](1, 1, 51)      '('
                        end_bracket_KeywordSegment:         [51](1, 1, 52)      ')'
                    [META] Dedent:                          [52](1, 1, 53)
                    end_bracket_KeywordSegment:             [52](1, 1, 53)      ')'
                [META] Dedent:                              [53](1, 1, 54)

```

**Comment by jonklippenstein-io** [comment]
It looks like parser/grammar.py L771-L780 is greedily consuming the whitespace into the Bracketed element since code_only is set to true.  Unfortunately I'm not familiar enough with the parser code to propose a good solution here.

**Comment by alanmcruickshank** [comment]
Thanks @jonklippenstein-io - looks like pretty accurate assessment. I'll have a look at this one, I think I need to revisit what the defaults are for whitespace consumption, or perhaps just mitigate this one in the rule definition.
